ballerina/tests/from_xml_test.bal

```diff
@@ -0,0 +1,750 @@
+
+    map<int[]> expected = {
+        age: [5]
+    };
+
+    map<int[]> actual = check fromXml(x);
+    test:assertEquals(actual, expected, msg = "testXmlToMapIntArray result incorrect");
+}
+
+@test:Config {
+    groups: ["fromXml"]
+}
+isolated function testXmlToMapFloatArray() returns error? {
+    var x = xml `<age>5</age>`;
+
+    map<float[]> expected = {
+        age: [5]
+    };
+
+    map<float[]> actual = check fromXml(x);
+    test:assertEquals(actual, expected, msg = "testXmlToMapFloatArray result incorrect");
+}
+
+@test:Config {
+    groups: ["fromXml"]
+}
+isolated function testXmlToMapDecimalArray() returns error? {
+    var x = xml `<age>5</age>`;
+
+    map<decimal[]> expected = {
+        age: [5]
+    };
+
+    map<decimal[]> actual = check fromXml(x);
+    test:assertEquals(actual, expected, msg = "testXmlToMapFloatArray result incorrect");
+}
+
+@test:Config {
+    groups: ["fromXml"]
+}
+isolated function testXmlToMapBooleanArray() returns error? {
+    var x = xml `<value>true</value>`;
+
+    map<boolean[]> expected = {
+        value: [true]
+    };
+
+    map<boolean[]> actual = check fromXml(x);
+    test:assertEquals(actual, expected, msg = "testXmlToMapBooleanArray result incorrect");
+}
+
+@test:Config {
+    groups: ["fromXml"]
+}
+isolated function testXmlToMapTable1() returns error? {
+    xml x1 = xml `<keys><key>value</key></keys>`;
+    map<table<map<string>>> expected = {"keys": table [{"key": "value"}]};
+    map<table<map<string>>> actual = check fromXml(x1);
+    test:assertEquals(actual, expected, msg = "testXmlToMapJson1 result incorrect");
+}
+
+@test:Config {
+    groups: ["fromXml"]
+}
+isolated function testXmlToMapTable2() returns error? {
+    xml x1 = xml `<keys><key>1</key></keys>`;
+    map<table<map<int>>> expected = {"keys": table [{key: 1}]};
+    map<table<map<int>>> actual = check fromXml(x1);
+    test:assertEquals(actual, expected, msg = "testXmlToMapTable2 result incorrect");
+}
+
+@test:Config {
+    groups: ["fromXml"]
+}
+isolated function testXmlToMapTable3() returns error? {
+    xml x1 = xml `<keys><key1>1</key1><key2>2</key2></keys>`;
+    map<table<map<int>>> expected = {keys: table [{key1: 1, key2: 2}]};
+    map<table<map<int>>> actual = check fromXml(x1);
+    test:assertEquals(actual, expected, msg = "testXmlToMapTable3 result incorrect");
+}
+
+type Table record {
+    int key1;
+    int key2;
+};
+
+@test:Config {
+    groups: ["fromXml"]
+}
+isolated function testXmlToMapTable4() returns error? {
+    xml x1 = xml `<keys><key1>1</key1><key2>2</key2></keys>`;
+    map<table<Table>> expected = {keys: table [{key1: 1, key2: 2}]};
+    map<table<Table>> actual = check fromXml(x1);
+    test:assertEquals(actual, expected, msg = "testXmlToMapTable4 result incorrect");
+}
+
+type BookStores11 record {
+    string ns0\:storeName;
+    int ns0\:postalCode;
+    boolean ns0\:isOpen;
+    Add ns0\:address;
+    Codes11 ns0\:codes;
+    @Attribute
+    string 'xmlns\:ns0;
```

Comment:
This can also define as `namespace` annotation right?

Reply:
Yes.